File: app/src/main/java/com/perflyst/twire/misc/ResizeWidthAnimation.java
package com.perflyst.twire.misc;

import android.view.View;
import android.view.animation.Animation;
import android.view.animation.Transformation;

/**
 * Created by SebastianRask on 24-01-2016.
 */
public class ResizeWidthAnimation extends Animation {
    private final int startWidth;
    private final int targetWidth;
    private final View view;

    public ResizeWidthAnimation(View view, int targetWidth) {
        this.view = view;
        this.targetWidth = targetWidth;
        startWidth = view.getWidth();
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        view.getLayoutParams().width = (int) (startWidth + (targetWidth - startWidth) * interpolatedTime);
        view.requestLayout();
    }

    @Override
    public void initialize(int width, int height, int parentWidth, int parentHeight) {
        super.initialize(width, height, parentWidth, parentHeight);
    }

    @Override
    public boolean willChangeBounds() {
        return true;
    }
}

File: app/src/main/java/com/perflyst/twire/misc/ResizeWidthAnimation.kt
package com.perflyst.twire.misc

import android.view.View
import android.view.animation.Animation
import android.view.animation.Transformation

/**
 * Created by SebastianRask on 24-01-2016.
 */
class ResizeWidthAnimation(private val view: View, private val targetWidth: Int) : Animation() {
    private val startWidth: Int = view.width

    override fun applyTransformation(interpolatedTime: Float, t: Transformation?) {
        view.layoutParams.width =
            (startWidth + (targetWidth - startWidth) * interpolatedTime).toInt()
        view.requestLayout()
    }

    override fun willChangeBounds(): Boolean {
        return true
    }
}
